src/foam/core/CSV.js

```diff
@@ -0,0 +1,359 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.csv',
+  name: 'Outputter',
+
+  documentation: 'CSV Outputter.',
+
+  properties: [
+    {
+      class: 'String',
+      name: 'buf_',
+      value: ''
+    },
+    {
+      class: 'Boolean',
+      name: 'includeQuotes',
+      help: 'If true, values are quoted with the `"` char',
+      value: true
+    },
+    {
+      class: 'String',
+      name: 'delimiter',
+      value: ','
+    },
+    {
+      class: 'String',
+      name: 'nestedObjectSeperator',
+      value: '__'
+    },
+    {
+      class: 'StringArray',
+      name: 'nestedObjectNames'
+    },
+    {
+      class: 'String',
+      name: 'nlStr',
+      value: '\n'
+    },
+    {
+      class: 'String',
+      name: 'undefinedStr',
+      value: ''
+    },
+    {
+      class: 'Boolean',
+      name: 'outputHeaderRow',
+      value: true
+    },
+    {
+      class: 'Function',
+      name: 'propertyPredicate',
+      value: function(o, p) { return ! p.transient; }
+    },
+    {
+      class: 'Boolean',
+      name: 'sortObjectKeys',
+      value: false
+    }
+  ],
+
+  methods: [
+    function out() {
+      for ( var i = 0 ; i < arguments.length ; i++ ) { 
+        this.buf_ += arguments[i];
+      }
+
+      return this;
+    },
+
+    function outputHeader(o) {
+      if ( ! this.outputHeaderRow ) return;
+
+      this.outputPropertyFilteredName(o);
+      this.removeTrailingComma().out(this.nlStr);
+    },
+
+    function removeTrailingComma() {
+      // Removes trailing comma
+      if ( this.buf_.charAt(this.buf_.length - 1) == ',' ) this.buf_ = this.buf_.slice(0, -1);
```

Comment:
This is very expensive. Instead, just output a , before each column except the first.